projects/packages/sync/src/class-sender.php

```diff
@@ -321,7 +321,28 @@ private function continue_full_sync_enqueue() {
 	 * @return boolean|WP_Error True if this sync sending was successful, error object otherwise.
 	 */
 	public function do_sync() {
-		return $this->do_sync_and_set_delays( $this->sync_queue );
+		if ( ! Settings::is_dedicated_sync_enabled() ) {
+			return $this->do_sync_and_set_delays( $this->sync_queue );
+		}
+
+		// If this is a dedicated Sync request, run Sync
+		// otherwise attempt to spawn a dedicated Sync request.
+		$is_dedicated_sync_request = Dedicated_Sender::is_dedicated_sync_request( $this->sync_queue );
```

Comment:
This is a performance issue/concern. The dedicated Sync Request should only perform sync actions. With this implementation it will trigger full processing of the site's main page, uncached as it is a post request. 

I think this should be simplified to 

```
if ( ! Settings::is_dedicated_sync_enabled() ) {
    return $this->do_sync_and_set_delays( $this->sync_queue );
} else {
    return Dedicated_Sender::spawn_sync( $this->sync_queue );
}
```
The is_dedicated_sync_request and logic to do sync needs to occur way up the call stack and exit when done no shutdown, or other hooks being fired.

Reply:
Excellent suggestion 👍  Gotcha!